core: Add volume controls to desktop and web

Korne127 opened the pull request:
This pull request adds volume controls to both desktop and web.
On desktop, they can be accessed through the menu bar, on web, through the right-click menu.
The volume controls contain a mute button and a slider from 0 to 100 in each.
The volume given to the player is adapted for logarithmic hearing.

On desktop, the volume controls are available, even if no SWF file is currently being played. When starting to play an SWF file, the current GUI volume controls are used.
Since the GUI volume controls use the PlayerOptions volume value as the default, this PR also fixes the issue that the PlayerOptions volume has previously not been adapted for logarithmic hearing.

The volume controls (and the options to open them) are multilingual. In this pull request, English and German translations have been added.

This closes #1771.

Dinnerbone commented:
> In this pull request, English and German translations have been added.

It'll be immediately deleted by the bot. Those files are generated, you need to provide translations on the crowdin website once the PR lands. Sorry 😅

Korne127 commented:
Oh haha, good to know; thanks for the information :)
I guess I'll also provide the translations on Crowdin then!

Korne127 commented:
Here are some videos that show how the volume controls look like (the German translations are used in this video, but the texts are "Volume settings" as the heading, "Mute" besides the checkbox and "Volume" besides the slider).

This is how it's used and looks like on desktop:

https://github.com/ruffle-rs/ruffle/assets/33464213/1736d3a6-64fd-48a4-a475-75a9c960298d


This is how it's used and looks like on web:

https://github.com/ruffle-rs/ruffle/assets/33464213/998735e8-d008-4755-bb17-11cb9d23d457



Note that on desktop, the volume is accessible even if no SWF is currently played, and it will be kept if you close the SWF and open another one (but not close Ruffle). Similarly, the volume on web is kept if you reload the SWF in the demo.

Croworbit commented:
This will be very handy!
all I can try to ask for is a visual like this
![volu](https://github.com/ruffle-rs/ruffle/assets/138729134/bf76d4f7-93bb-4924-b07b-def2f882fcd4)

Dinnerbone commented:
> This will be very handy! all I can try to ask for is a visual like this ![volu](https://user-images.githubusercontent.com/138729134/261874110-bf76d4f7-93bb-4924-b07b-def2f882fcd4.png)

That'd be pretty cool, but maybe in the future - this is already a big improvement :)

Croworbit commented:
you're right, this would better fit a future PR and not this one!

Korne127 commented:
@Croworbit I am not the biggest frontend developer, so while I was able to create a GUI like this, I probably won't be able to include special animations or stuff like that.
While I'm actually pretty confident with how the volume controls look now, if someone wants, they're welcome to improve it after this PR gets merged.

Korne127 commented:
I updated the PR and changed two things:
1. The volume text next to the slider is now also greyed out on web if the mute checkbox is checked. When taking the videos, I noticed that this was different on web and on desktop (now it's consistent) and it looks better this way.
2. I moved the logarithmic volume transformation into the AudioMixer (as suggested by @adrian17).
